src/buffer/out/textBuffer.cpp

```diff
@@ -1832,6 +1895,43 @@ const TextBuffer::TextAndColor TextBuffer::GetText(const bool includeCRLF,
     return data;
 }
 
+size_t TextBuffer::SpanLength(const til::point coordStart, const til::point coordEnd) const
+{
+    const auto bufferSize = GetSize();
+    // The coords are inclusive, so to get the (inclusive) length we add 1.
+    const auto length = bufferSize.CompareInBounds(coordEnd, coordStart) + 1;
+    return gsl::narrow<size_t>(length);
+}
+
+// Routine Description:
+// - Retrieves the plain text data between the specified coordinates.
+// Arguments:
+// - trimTrailingWhitespace - remove the trailing whitespace at the end of the result.
+// - start - where to start getting text (should be at or prior to "end")
+// - end - where to end getting text
+// Return Value:
+// - Just the text.
+std::wstring TextBuffer::GetPlainText(const til::point& start, const til::point& end) const
+{
+    std::wstring text;
+    auto spanLength = SpanLength(start, end);
+    text.reserve(spanLength);
+
+    auto it = GetCellDataAt(start);
+
+    for (; it && spanLength > 0; ++it, --spanLength)
+    {
+        const auto& cell = *it;
+        if (!cell.DbcsAttr().IsTrailing())
+        {
+            const auto chars = cell.Chars();
+            text.append(chars);
+        }
+    }
+
+    return text;

```

Comment:
Oh damn... That's a good point. I should've scrolled down first. 😅
On one hand I find this suggestion brilliant (less code duplication), but on the other the horrendous overhead from both `GetTextRects` and `GetText` is triggering me a bit. 😄

In case that this works correctly, I'd like to "undo" my above comment regarding the use of `GetTextSpans`. Even assuming that we switch from selection rects over to selection spans in the future, it's still probably for the best if we have less code to replace by then. 🙂